util/time_accounting.h

```diff
@@ -0,0 +1,50 @@
+/*
+   Copyright 2019 Bloomberg Finance L.P.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+#ifndef _time_accounting_h
+#define _time_accounting_h
+
+
+#ifndef NDEBUG
+enum { CHR_IXADDK, CHR_DATADD, CHR_TMPSVOP, CHR_MAX };
+
+#define CHRONO_START()   \
+{                        \
+    struct timeval __tv; \
+    gettimeofday(&__tv, NULL);
```

Comment:
gettimeofday isn't cheap. We should at least not do this by default.

Reply:
It is off by default. On only if DEBUG is on.

Reply:
Gotcha. This can be done by just 1 macro though. For example, you'd have a macro like the following:

```c
#define TIMER(code) do { struct timeval __tv; gettimeofday(&__tv, NULL); X; accumulate_time(name, chrono_stop(&__tv)); } while (0)
```

And you'd time a code block like this:

```c
TIMER(
    // line 1
    // line 2
    // line 3
)
```